### ISSUE TITLE
Navigation to next same and different styles on a page with Foto in a pdf document: NVDA throws a list index error and the offset is too big for the buffer, causing NVDA to be silent.

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#16407]
### Steps to reproduce:
1. Open NVDA
2. Assign [ and shift+[ for different style navigation 
3. Assign ] and shift+] for same style navigation
4. Open the attached pdf document with adobe Reader
5. Switch to a page which contains only a foto, e.g. page one, two or four
[annual-report-2022-en.pdf](https://github.com/nvaccess/nvda/files/15002529/annual-report-2022-en.pdf)
6. Press the assigned commands for navigating to next and previous text styles

### Actual behavior:
NVDA gives a list index error, and the offset is too big for the buffer when attempting to navigate to next different or next same style text.
```
IO - inputCore.InputManager.executeGesture (23:01:47.223) - winInputHook (10508):
Input: kb(laptop):ü
DEBUGWARNING - RPC process 23548 (Acrobat.exe) (23:01:47.223) - Dummy-11 (1624):
Thread 20588, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::getLineOffsets, 1068:
Offset of 0 too big for buffer, returning false

ERROR - scriptHandler.executeScript (23:01:47.223) - MainThread (19244):
error executing script: <bound method BrowseModeTreeInterceptor.addQuickNav.<locals>.<lambda> of <virtualBuffers.adobeAcrobat.AdobeAcrobat object at 0x0A20F190>> with gesture 'ü'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 295, in executeScript
  File "browseMode.pyc", line 558, in <lambda>
  File "browseMode.pyc", line 516, in _quickNavScript
  File "browseMode.pyc", line 2395, in _iterTextStyle
IndexError: list index out of range
IO - inputCore.InputManager.executeGesture (23:01:47.678) - winInputHook (10508):
Input: kb(laptop):shift+ü
IO - speech.speech.speak (23:01:47.679) - MainThread (19244):
Speaking ['No previous different style text']
IO - inputCore.InputManager.executeGesture (23:01:48.404) - winInputHook (10508):
Input: kb(laptop):plus
DEBUGWARNING - RPC process 23548 (Acrobat.exe) (23:01:48.404) - Dummy-11 (1624):
Thread 20588, build\x86_64\vbufBase\storage.cpp, VBufStorage_buffer_t::getLineOffsets, 1068:
Offset of 0 too big for buffer, returning false

ERROR - scriptHandler.executeScript (23:01:48.404) - MainThread (19244):
error executing script: <bound method BrowseModeTreeInterceptor.addQuickNav.<locals>.<lambda> of <virtualBuffers.adobeAcrobat.AdobeAcrobat object at 0x0A20F190>> with gesture 'plus'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 295, in executeScript
  File "browseMode.pyc", line 558, in <lambda>
  File "browseMode.pyc", line 516, in _quickNavScript
  File "browseMode.pyc", line 2395, in _iterTextStyle
IndexError: list index out of range
IO - inputCore.InputManager.executeGesture (23:01:48.733) - winInputHook (10508):
Input: kb(laptop):shift+plus
IO - speech.speech.speak (23:01:48.733) - MainThread (19244):
Speaking ['No previous same style text']
```

### Expected behavior:
NVDA should report "no next different style text" or "no next same style text". This works as expected for previous styles commands.

### NVDA logs, crash dumps and other attachments:
n/a

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
Version: alpha-31612,3d4aOa8f (2024.2.0.31612)

#### Windows version:
Windows 11 23 H2

#### Name and version of other software in use when reproducing the issue:
Adobe acrobat reader Continuous Release I Version 2024.002.20687 1 64-Bit

#### Other information about your system:
n/a

#### Does the issue still occur after restarting your computer?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
n/a

#### If NVDA add-ons are disabled, is your problem still occurring?
yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by Adriani90** [comment]
cc: @mltony

**Comment by seanbudd** [comment]
@mltony - we may need to revert this feature over this issue if this issue isn't fixed in beta

**Comment by mltony** [comment]
This is fixed in #16430.
